uber/config.py

```diff
@@ -671,14 +720,25 @@ def ADMIN_WRITE_ACCESS_SET(self):
     @cached_property
     def ADMIN_PAGES(self):
```

Comment:
This now dynamically builds a list of public site sections by looking for the `public` attribute. This does mean I had to add `@public` to all the public site sections, as all_renderable doesn't add the public attribute until its child functions are actually called.

Reply:
Instead of having to add ``@public`` to all of the public site-sections, I would recommend modifying ``all_renderable`` instead such that in its ``__call__`` we say

```python
    def __call__(self, klass):
        if self.public:
            klass = public(klass)
        for name, func in klass.__dict__.items():
            ...
```

That will accomplish the same thing but without needing to specify ``@public`` before all of the public site sections, which feels redundant since we're already saying ``public=True`` one line later in all of those cases.

Reply:
Wait, would that actually work with the way we're iterating through the methods? I thought since we inspect them that wouldn't call this bit of code... if that can work, that's definitely better!

Reply:
Did this and it worked! Merging now.

Reply:
Yeah, the reason why it works is that the ``__call__`` is invoked on the whole class at import time, so this causes us to set the ``.public`` attribute at import time rather than when the methods being decorated are called at runtime.